common/database/mongo_database.go

```diff
@@ -205,6 +206,20 @@ func (db *MongoDatabase) UpdateAll(collection_name string, selector interface{},
 	return &change_results, convertMgoError(err)
 }
 
+/*
+	Finds an item based on the given selector and patches it with the data in update
+*/
+func (db *MongoDatabase) Patch(collection_name string, selector interface{}, update *map[string]interface{}) error {
+	current_session := db.GetSession()
+	defer current_session.Close()
+
+	collection := current_session.DB(db.name).C(collection_name)
+
+	// Use mongodb set operator to update only the provided fields.
+	err := collection.Update(selector, bson.M{"$set": update})
```

Comment:
It's my understanding that this will only merge top-level changes, not nested fields. If so, it might be worth noting this in the comment above.

ie

```
{
  "foo": {
    "bar1": 1,
  }
}
```

merged with

```
{
  "foo": {
    "bar2": 2,
  }
}
```

results in

```
{
  "foo": {
    "bar2": 2,
  }
}
```